application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,245 @@
+    }
+
+    /**
+     * Method for adding the channel that the status will be printed in. Even though the method only
+     * stores the long id it requires, the method requires the actual {@link TextChannel} to be
+     * passed because it needs to verify it as well as store the guild id.
+     *
+     * This method also calls a method which updates the status of the channels in the {@link Guild}
+     * . So always add the status channel AFTER you have added all monitored channels for the guild,
+     * see {@link #addChannelToMonitor(long)}.
+     *
+     * @param channel the channel the status message must be displayed in
+     */
+    public void addChannelForStatus(@NotNull final TextChannel channel) {
+        postStatusInChannel.put(channel.getGuild().getIdLong(), channel.getIdLong());
+        updateStatusFor(channel.getGuild());
+    }
+
+    /**
+     * This method tests whether a guild id is configured for monitoring in the free command system.
+     * 
+     * @param guildId the id of the guild to test.
+     * @return {@code true} if the guild is configured in the system, {@code false} otherwise.
+     */
+    public boolean isMonitoringGuild(final long guildId) {
+        return postStatusInChannel.containsKey(guildId);
+    }
+
+    /**
+     * This method tests whether a channel id is configured for monitoring in the free command
+     * system.
+     * 
+     * @param channelId the id of the channel to test.
+     * @return {@code true} if the channel is configured in the system, {@code false} otherwise.
+     */
+    public boolean isMonitoringChannel(final long channelId) {
+        return channelsToMonitor.containsKey(channelId);
+    }
+
+    /**
+     * This is a delegation method to pass operations to {@link ChannelStatus}. see
+     * {@link ChannelStatus#isBusy()} for details.
+     *
+     * @param channelId the id for the channel to test.
+     * @return {@code true} if the channel is 'busy', false if the channel is 'free'.
+     */
+    public boolean isChannelBusy(final long channelId) {
+        return channelsToMonitor.get(channelId).isBusy();
+    }
+
+    /**
+     * This method tests if a channel is currently active by fetching the latest message and testing
+     * if it was posted more than an hour ago.
+     *
+     * @param channel the channel to test.
+     * @return {@code true} if the channel is inactive, false if it has received messages more
+     *         recently than an hour ago.
+     */
+    public boolean isChannelInactive(@NotNull final TextChannel channel) {
+        if (!channelsToMonitor.containsKey(channel.getIdLong())) {
+            throw new IllegalArgumentException(
+                    "Channel requested %s is not monitored by free channel"
+                        .formatted(channel.getName()));
+        }
+        // todo change the entire inactive test to work via restactions
+        return channel.getHistory()
```

Comment:
Just an idea, this is fine too.

But you could make use MessageChannel#getLatestMessageIdLong, and extract the time created from there?

If you send a message and delete it, it'll still be counted as the latest message.

So maybe not what you want

Reply:
Correct, not what we want.
You are the one who told me that even if a message gets deleted it will still count as the latest message and trying to retrieve it results in `null`. (I actually ran into that problem by accident because i was deleting testing messages)

If a person does post something and then deletes it (firstly we cant then determine what time it happened at and secondly we dont want to leave a channel open if ppl keep on posting and deleting messages without the conversation progressing in any way.

Reply:
You don't need to retrieve the message is the time is in its ID, and sure, I get that.

Reply:
so u suggest getLatestMessageID() Snowflake -> get time the id because its all local data rather than getMessageHistory, even though deleted messages will still reflect as latest message?

Reply:
Because it includes deleted messages, it's your decision what you want to use.